bin/si/src/main.rs

```diff
@@ -37,18 +43,25 @@ async fn main() -> Result<()> {
         std::path::Path::new(&std::env::var("HOME")?)
             .join(".docker")
             .join("run")
-            .join("docker.sock"),
-        std::path::Path::new("/var/run/docker.sock").to_path_buf(),
+            .join("docket.sock"),
+        // std::path::Path::new("/var/run/docker.sock").to_path_buf(),
     ];
 
-    let docker_socket = docker_socket_candidates
-        .iter()
-        .find(|candidate| candidate.exists())
-        .ok_or(eyre!(
-            "failed to determine Docker socket location; candidates={docker_socket_candidates:?}"
+    let docker: DockerClient;
+    if let "" = docker_sock.as_str() {
+        let socket = docker_socket_candidates
+            .iter()
+            .find(|candidate| candidate.exists())
+            .ok_or(eyre!(
+            "failed to determine Docker socket location. Set a custom location using `--docker_sock` \
```

Comment:
Pretty sure clap will turn that into `--docker-sock` (underscores become dashes)

Reply:
you are correct!